### PR TITLE
[WIP] Improve query performance with read a head blocklet and changed number compressor.

### PR DESCRIPTION
**Pull request by ravipesala**
Need to work out with backward compatability

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/757/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/759/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/761/

File: core/src/main/java/org/apache/carbondata/core/keygenerator/mdkey/NumberCompressor.java [hunk 1]
```diff
@@ -137,12 +161,31 @@ public NumberCompressor(int cardinality) {
...
   public int[] unCompress(byte[] key, int offset, int length) {
-    int ls = length;
-    int arrayLength = (ls * BYTE_LENGTH) / bitsLength;
-    long[] words = new long[getWordsSizeFromBytesSize(ls)];
-    unCompressVal(key, ls, words, offset);
-    return getArray(words, arrayLength);
+    int len = length / bytesPerKey;
+    if (length % bytesPerKey != 0) {
+      throw new IllegalArgumentException(
+          "Some thing wrong while decompress" + length + " " + bytesPerKey);
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/keygenerator/mdkey/NumberCompressor.java]: Both length is read from footer, right? If comes here, mean the file is corrupted.

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/762/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/790/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/801/

**Comment by ravipesala** [comment]
Same has been handled in https://github.com/apache/incubator-carbondata/pull/584, so closing this PR.

**Pull request closed without merging**